core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -349,7 +380,7 @@ public void cancel(@Nullable String message, @Nullable Throwable cause) {
     try {
       // Cancel is called in exception handling cases, so it may be the case that the
       // stream was never successfully created or start has never been called.
-      if (stream != null) {
+      if (streamStarted) {
```

Comment:
I don't think this change is equivalent. `stream != null` means the user hasn't called start() at all. Once they call start, calling cancel must do something; at the very least it should notify the user's listener.